definitions.h

```diff
@@ -47,6 +52,327 @@ struct Vertex
     double w;
 };
 
+class Matrix
+{
+    public:
+		Matrix();
+        Matrix(int c, int r);
+        void insert(int x, int y, double data);
+        double returnVal(int x, int y);
+		int getColumnSize();
+		int getRowSize();
+		int getSize();
+		//Matrix matrixCopy(Matrix & m);
+		//Have to write 4 different const types
+		Matrix operator*(const Matrix & m) const;
+		//Matrix operator*(const Vertex & m) const;
+		Vertex operator*(const Vertex & m) const;
+		Matrix& operator=(Matrix const& m);
+		Matrix & operator=(std::initializer_list<double> m);
+		Matrix & operator=(std::initializer_list<std::initializer_list<double>> m);
+
+    private:
+        int column;
+        int row;
+		int size;
+        double matrixData[16];
+};
+Matrix::Matrix()
+{
+	column = 0;
+    row = 0;
+	size = 0;
+}
+
+Matrix::Matrix (int c, int r)
+{
+    column = c;
+    row = r;
+	size = row * column;
+	for(int i = 0; i < size; i++)
+	{
+		matrixData[i] = 0;
+	}
+}
+
+void Matrix::insert(int x, int y, double data)
+{
+    matrixData[(x * column) + y] = data;
+}
+
+double Matrix::returnVal(int x, int y)
+{
+    return matrixData[(x * column) + y];
+}
+
+int Matrix::getColumnSize()
+{
+	return column;
+}
+
+int Matrix::getRowSize()
+{
+	return row;
+}
+
+int Matrix::getSize()
+{
+	return size;
+}
+
+Matrix computeDimensions(int c1, int r1, int c2, int r2)

```

Comment:
You went in-depth in your implementation of the Matrix class.  I would like to ask you how all this works sometime.